uat/custom-components/client-java-sdk/src/main/java/com/aws/greengrass/testing/mqtt5/client/sdkmqtt/MqttConnectionImpl.java

```diff
@@ -467,15 +474,14 @@ private Mqtt5Client createClient(MqttLib.ConnectionParams connectionParams) {
                 .withSessionBehavior(clientSessionBehavior)
                 .withPort(Long.valueOf(connectionParams.getPort()))
                 .withLifeCycleEvents(lifecycleEvents)
-                .withPublishEvents(publishEvents);
+                .withPublishEvents(publishEvents)
+                .withMaxReconnectDelayMs(RECONNECT_DELAY_MS)
+                .withMinReconnectDelayMs(RECONNECT_DELAY_MS);
```

Comment:
We did that in https://github.com/aws-greengrass/aws-greengrass-client-device-auth/pull/363/files#diff-07a8fb9736f95fa43b3247de8f7c32b197957c7d236f2f59200538877cb21906R274

Reply:
So why is this needed then? Instead of changing the reconnect time here and mosquitto why don't you use the disconnect method?

Reply:
That code just configure options of future connection. In some clients connect is asynchronous. Can't disconnect before connection is established. But even when library return failure it try to reconnect with default settings. Here I just rewrite default setting of reconnect delay to 1 day to avoid reconnect. Despire of Paho Java SDK has no method to disable reconnect feature. stop() is also used, but large reconnect delay avoid race before stop().

Reply:
There should not be a race between stop and reconnect. You should call stop from the connection failure callback. This executes in a single thread

Reply:
I called stop(). Here I just effectively disable reconnect feature of SDK MQTT client